### PR TITLE
Fix showing wrong method name in completions for methods in abstract objects

### LINKED ISSUE #35002: Abstract objects gives incorrect VSCode suggestions
**Description:**
Ballerina time module has an implementation, which uses an abstract object `Zone`. Then, we extend the abstract object to implement the actual logic `TimeZone`.

For these APIs, the VSCode gives incorrect suggestions, see below.

Screenshot 2022-02-10 at 11 57 38

The code snippet it gives is as follows,
Screenshot 2022-02-10 at 12 13 40

But the correct suggestion should be,

```ballerina
import ballerina/time;
import ballerina/io;

public function main() returns error? {
    time:Zone timezone = check time:loadSystemZone();
    time:Civil civil = timezone.utcToCivil(time:utcNow()); // without Zone
    io:println(civil);
}

```
**Affected Versions:**
SL GA

**OS, DB, other environment details and versions:**
- macOS
- VSCode version: 1.64.1
- Ballerina VSCode extension: 3.0.0

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #35002 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java [hunk 1]
```diff
@@ -1646,8 +1646,7 @@ private void populateIntersectionTypeReferencedFunctions(DataInputStream inputSt
                 BInvokableType attachedFuncType = (BInvokableType) readTypeFromCp();
                 Name funcName = names.fromString(Symbols.getAttachedFuncSymbolName(
                         objectSymbol.name.value, attachedFuncName));
-                Name funcOrigName = names.fromString(Symbols.getAttachedFuncSymbolName(
-                        objectSymbol.originalName.value, attachedFuncName));
+                Name funcOrigName = names.fromString(attachedFuncName);
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java:1649]: There's a slight issue here in the BIR. It seems we are not writing the original name to the BIR as we do for regular function definitions. For regular functions defs: https://github.com/ballerina-platform/ballerina-lang/blob/a8f28183cee016fab227886b95aa3047315b5eb2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java#L208

For attached functions, not writing the original name: https://github.com/ballerina-platform/ballerina-lang/blob/a8f28183cee016fab227886b95aa3047315b5eb2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L493

**Review by pubudu91 — COMMENTED** [review]
If making a change to the BIR format, best send this fix to `master`. Also, @MaryamZi FYI

**Review by pubudu91 — DISMISSED** [review]
Changes LGTM. If the FE team is also ok with these BIR changes, let's merge this. @gimantha FYI.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java [hunk 2]
```diff
@@ -25,9 +25,9 @@
 
     public static final int MAGIC_NUMBER = 0xBA1DA4CE;
     public static final short VERSION_NUMBER = 50;
-    public static final int BIR_VERSION_NUMBER = 64;
-    public static final short MIN_SUPPORTED_VERSION = 64;
-    public static final short MAX_SUPPORTED_VERSION = 64;
+    public static final int BIR_VERSION_NUMBER = 65;
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: @pubudu91 wouldn't this have an impact on modules built on previous patch releases? (for example, if an external user had published a module to central with a previous patch version of swan lake, would that module work with the next patch release?)

Comment (pubudu91) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Ah sorry, this PR should go to master. Not the patch branch. We are doing BIR changes in master right? @dulajdilshan please note.

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Changed

**Comment by mohanvive** [comment]
@dulajdilshan Can you please rebase this with the master agin?

**Comment by mohanvive** [comment]
@dulajdilshan Can you please rebase this with the master agin?

**Review by mohanvive — APPROVED** [review]

**Comment by dulajdilshan** [comment]
> @dulajdilshan Can you please rebase this with the master agin?

This is PR is up to date with the master

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java (hunk 2 above)
Comment (gimantha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Still wouldnt this be a major change for someone who is migrating to update1?

Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Do you mean the same as your first comment, but from patch to updates? We get only the source from central.

I think the BIR will/should just get regenerated. @hasithaa mentioned some time back that there was a discussion on if we can do BIR changes in updates and that the conclusion was that we can. Looping in @azinneera and @hevayo also.

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Only the source is shared. We have a cache directory per distribution. So a new distribution will create its own bir and jar.

E.g: 
Dist 2201.0.1 cache location: `~/.ballerina/repositories/central.ballerina.io/cache-2201.0.1`
Dist 2201.0.2 cache location: `~/.ballerina/repositories/central.ballerina.io/cache-2201.0.2`

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**